src/libostree/ostree-repo-commit.c

```diff
@@ -2237,14 +2237,24 @@ _ostree_repo_commit_modifier_apply (OstreeRepo               *self,
   OstreeRepoCommitFilterResult result;
   GFileInfo *modified_info;
 
-  if (modifier == NULL || modifier->filter == NULL)
+  if (modifier == NULL ||
+      (modifier->filter == NULL &&
+       (modifier->flags & OSTREE_REPO_COMMIT_MODIFIER_FLAGS_CANONICAL_PERMISSIONS) == 0))
     {
       *out_modified_info = g_object_ref (file_info);
       return OSTREE_REPO_COMMIT_FILTER_ALLOW;
     }
 
   modified_info = g_file_info_dup (file_info);
   result = modifier->filter (self, path, modified_info, modifier->user_data);
+
+  if ((modifier->flags & OSTREE_REPO_COMMIT_MODIFIER_FLAGS_CANONICAL_PERMISSIONS) != 0 &&
+      g_file_info_get_file_type (file_info) == G_FILE_TYPE_REGULAR)
+    {
+      guint current_mode = g_file_info_get_attribute_uint32 (modified_info, "unix::mode");
+      g_file_info_set_attribute_uint32 (modified_info, "unix::mode", current_mode | 0744);
```

Comment:
We *could* make this automatically mean uid/gid 0 and no-xatts too, but that means you can't do the indiviual operations separately